spec/deployment_spec.rb

```diff
@@ -114,3 +116,203 @@
+                  expect(certificate['spec']['dnsNames'].count).to eq(1)
+                  expect(certificate['spec']['dnsNames'][0]).to eq("#{@name}.#{Config.domain}")
+
+                  expect(certificate['status']).to_not be_nil
+                  expect(certificate['status']['conditions']).to_not be_nil
+                  expect(certificate['status']['conditions'].count).to eq(1)
+                  expect(certificate['status']['conditions'][0]['type']).to eq('Ready')
+                  expect(certificate['status']['conditions'][0]['status']).to eq('True')
+
+                  expect(Time.parse(certificate['status']['notAfter']) > (Time.now + 60*60*24*5)).to eq(true)
+                  expect(Time.parse(certificate['status']['notAfter']) < (Time.now + 60*60*24*180)).to eq(true)
+                  expect(Time.parse(certificate['status']['notBefore']) > (Time.now - 60*60*24*1)).to eq(true)
+                }
+              end
+
+              it "can be https queried via domain [#{Config.domain}]" do
+                wait_until(30,5) {
+                  response = https_get("https://#{@name}.#{Config.domain}/hi")
+                  expect(response).to_not be_nil
+                  expect(response.code).to eq(200)
+                  expect(response.headers[:content_type]).to include('text/plain')
+                  expect(response.body).to eq('Hello')
+                }
+              end
+
+              it "can validate the app config via domain [#{Config.domain}]" do
+                wait_until(30,5) {
+                  response = https_get("https://#{@name}.#{Config.domain}/config")
+                  expect(response).to_not be_nil
+                  expect(response.code).to eq(200)
+                  expect(response.headers[:content_type]).to include('application/x-yaml')
+                  expect(response.body).to include('fileServerFolder: /var/www')
+                }
+              end
+
+              context 'with files created on the persistent volume' do
+                before(:all) do
+                  response = https_patch("https://#{@name}.#{Config.domain}/greet/howdy", "")
+                  expect(response.code).to eq(202)
+                end
+
+                it "can serve files from the persistent volume" do
+                  wait_until(30,5) {
+                    for file_suffix in 0..10 do
+                      response = https_get("https://#{@name}.#{Config.domain}/static/greet-#{file_suffix}.txt")
+                      expect(response).to_not be_nil
+                      expect(response.code).to eq(200)
+                      expect(response.headers[:content_type]).to include('text/plain')
+                      expect(response.body).to include('Howdy')
+                    end
+                  }
+                end
+              end
+            end
+
+          else # no lets-encrypt, let's just try with HTTP
+            it "can be http queried via domain [#{Config.domain}]" do
```

Comment:
calls  of /hi and /config endpoints can be consolidated into one single test (only /config would be enough), no need for 2 tests.